game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorDialog.java

```diff
@@ -46,14 +46,11 @@
    * Shows the Odds Calculator dialog and initializes it using the current state of the specified
    * territory.
    */
-  public static void show(final TripleAFrame taFrame, final Territory t, final History history) {
-    // Note: The history param may not be the same as taFrame.getGame().getData().getHistory() as
-    // GameData
-    // gets cloned when showing history with a different History instance that doesn't correspond to
-    // what's
-    // shown.
-    final BattleCalculatorPanel panel =
-        new BattleCalculatorPanel(taFrame.getGame().getData(), history, taFrame.getUiContext(), t);
+  public static void show(final TripleAFrame taFrame, final Territory t, final GameData data) {
+    // Note: The data param may not be the same as taFrame.getGame().getData() as GameData gets
+    // cloned when showing history with a different History instance that doesn't correspond to
+    // what's shown.
```

Comment:
I think this comment probably could be removed. I'd opt to put a mention in the javadoc that we use the 'data' param to init the battle calculator. Then it's clear we are not using the game data from 'taFrame'